config/envoyconfig/listeners.go

```diff
@@ -668,42 +668,47 @@ func (b *Builder) buildDownstreamTLSContext(ctx context.Context,
 	}
 
 	envoyCert := b.envoyTLSCertificateFromGoTLSCertificate(ctx, cert)
-	return &envoy_extensions_transport_sockets_tls_v3.DownstreamTlsContext{
+	dtc := &envoy_extensions_transport_sockets_tls_v3.DownstreamTlsContext{
 		CommonTlsContext: &envoy_extensions_transport_sockets_tls_v3.CommonTlsContext{
-			TlsParams:             tlsParams,
-			TlsCertificates:       []*envoy_extensions_transport_sockets_tls_v3.TlsCertificate{envoyCert},
-			AlpnProtocols:         alpnProtocols,
-			ValidationContextType: b.buildDownstreamValidationContext(ctx, cfg, domain),
+			TlsParams:       tlsParams,
+			TlsCertificates: []*envoy_extensions_transport_sockets_tls_v3.TlsCertificate{envoyCert},
+			AlpnProtocols:   alpnProtocols,
 		},
 	}
+	if clientCA := clientCAForDomain(cfg, domain); len(clientCA) > 0 {
+		dtc.CommonTlsContext.ValidationContextType = b.buildDownstreamValidationContext(ctx, cfg, clientCA)
+		dtc.RequireClientCertificate = wrapperspb.Bool(true)
+	}
+	return dtc
 }
 
-func (b *Builder) buildDownstreamValidationContext(ctx context.Context,
-	cfg *config.Config,
-	domain string,
-) *envoy_extensions_transport_sockets_tls_v3.CommonTlsContext_ValidationContext {
-	needsClientCert := false
-
-	if ca, _ := cfg.Options.GetClientCA(); len(ca) > 0 {
-		needsClientCert = true
-	}
-	if !needsClientCert {
-		for _, p := range getPoliciesForDomain(cfg.Options, domain) {
-			if p.TLSDownstreamClientCA != "" {
-				needsClientCert = true
-				break
-			}
+// clientCAForDomain returns a bundle of all per-route client CAs configured
+// for the given domain, or else the globally configured client CA.
+func clientCAForDomain(cfg *config.Config, domain string) []byte {
+	var bundle []byte
+	for _, p := range getPoliciesForDomain(cfg.Options, domain) {
```

Comment:
if the user configured multiple routes per domain with different `tls_downstream_client_ca` options, this should return an invalid configuration error

Reply:
as discussed, making a bundle would work.

Reply:
So far in my manual testing this "bundle" approach appears to be working as intended:
 - I configure two routes for the same domain, each with a different CA
 - I get a browser prompt that lets me select certificates from either CA
 - selecting a certificate from the first route's CA lets me through to the login page
 - selecting a certificate from the second route's CA gives me a Pomerium 495 error page

Reply:
I've added an integration test case to exercise this behavior: [TestMultipleDownstreamClientCAs](https://github.com/pomerium/pomerium/pull/4228/files#diff-0eb2b3039ed5088bb153887ae325c6cf4c8ec35063b236524fd1247bc571c5a2R399).